lib/defaults.js

```diff
@@ -0,0 +1,152 @@
+      globalPrefix = path.dirname(process.execPath);
+    } else {
+      // /usr/local/bin/node --> prefix=/usr/local
+      globalPrefix = path.dirname(path.dirname(process.execPath)); // destdir only is respected on Unix
+
+      if (process.env.DESTDIR) {
+        globalPrefix = path.join(process.env.DESTDIR, globalPrefix);
+      }
+    }
+
+    defaults = {
+      access: null,
+      'allow-same-version': false,
+      'always-auth': false,
+      also: null,
+      'auth-type': 'legacy',
+      'bin-links': true,
+      browser: null,
+      ca: null,
+      cafile: null,
+      cache: cache,
+      'cache-lock-stale': 60000,
+      'cache-lock-retries': 10,
+      'cache-lock-wait': 10000,
+      'cache-max': Infinity,
+      'cache-min': 10,
+      cert: null,
+      color: true,
+      depth: Infinity,
+      description: true,
+      dev: false,
+      'dry-run': false,
+      editor: osenv.editor(),
+      'engine-strict': false,
+      force: false,
+      'fetch-retries': 2,
+      'fetch-retry-factor': 10,
+      'fetch-retry-mintimeout': 10000,
+      'fetch-retry-maxtimeout': 60000,
+      git: 'git',
+      'git-tag-version': true,
+      global: false,
+      globalconfig: path.resolve(globalPrefix, 'etc', 'npmrc'),
+      'global-style': false,
+      group: process.platform === 'win32' ? 0 : process.env.SUDO_GID || process.getgid && process.getgid(),
+      'ham-it-up': false,
+      heading: 'npm',
+      'if-present': false,
+      'ignore-scripts': false,
+      'init-module': path.resolve(home, '.npm-init.js'),
+      'init-author-name': '',
+      'init-author-email': '',
+      'init-author-url': '',
+      'init-version': '1.0.0',
+      'init-license': 'ISC',
+      json: false,
+      key: null,
+      'legacy-bundling': false,
+      link: false,
+      'local-address': undefined,
+      loglevel: 'warn',
+      logstream: process.stderr,
+      'logs-max': 10,
+      long: false,
+      maxsockets: 50,
+      message: '%s',
+      'metrics-registry': null,
+      'node-version': process.version,
+      'onload-script': false,
+      only: null,
+      optional: true,
+      parseable: false,
+      prefix: globalPrefix,
+      production: process.env.NODE_ENV === 'production',
+      'progress': !process.env.TRAVIS && !process.env.CI,
+      'proprietary-attribs': true,
+      proxy: null,
+      'https-proxy': null,
+      'user-agent': 'npm/{npm-version} ' + 'node/{node-version} ' + '{platform} ' + '{arch}',
+      'rebuild-bundle': true,
+      registry: 'https://registry.npmjs.org/',
+      rollback: true,
+      save: false,
+      'save-bundle': false,
+      'save-dev': false,
+      'save-exact': false,
+      'save-optional': false,
+      'save-prefix': '^',
+      scope: '',
+      'scripts-prepend-node-path': 'warn-only',
+      searchopts: '',
+      searchexclude: null,
+      searchlimit: 20,
+      searchstaleness: 15 * 60,
+      'send-metrics': false,
+      shell: osenv.shell(),
```

Comment:
I think just traversing the object with the parser and comparing it to another object we've defined with the values we want to replace is the easiest option, and also the best.